js/src/api/contract/contract.js

```diff
@@ -79,15 +81,18 @@ export default class Contract {
     return this;
   }
 
-  deploy (code, values) {
-    const options = {
-      data: code,
-      gas: 900000
-    };
+  deploy (optionsOrCode, values) {
+    const options = Object.assign(
+      { gas: DEFAULT_DEPLOY_GAS },
+      isObject(optionsOrCode)
+        ? optionsOrCode
+        : { data: optionsOrCode }
+    );
 
     return this._api.eth
       .postTransaction(this._encodeOptions(this.constructors[0], options, values))
-      .then((txhash) => this.pollTransactionReceipt(txhash))
+      .then(this._pollCheckRequest)
+      .then(this._pollTransactionReceipt)
```

Comment:
It might be a good idea to wait for at least 1-2 confirmations - Parity (without `--geth` mode) will return a receipt even if transaction is in pending block (which means you can't really interact with the contract yet (and `getCode` will return `0x0`))